### PR TITLE
Create SmartUtility.Core and SmartUtility.Functions projects

### PR DESCRIPTION
**Pull request by stansw**

### CODE REVIEW

File: SmartUtilityAccelerator/SmartUtility/SmartUtility.Core/Sonnen/Client/Generated/SonnenBatterieClient.cs [hunk 1]
```diff
@@ -17,7 +17,7 @@ namespace Sonnen.Client
     using System.Threading;
     using System.Threading.Tasks;
 
-    public partial class SonnenClient : ServiceClient<SonnenClient>, ISonnenClient
+    public partial class SonnenBatterieClient : ServiceClient<SonnenBatterieClient>, ISonnenBatterieClient
```
Comment (conork20) [inline SmartUtilityAccelerator/SmartUtility/SmartUtility.Core/Sonnen/Client/Generated/SonnenBatterieClient.cs]: I would use SonnenSolarClient or SonnenClient here, which is a little more generic.

File: SmartUtilityAccelerator/SmartUtility/SmartUtility.Core/Sonnen/SonnenBatterieBridge.cs [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+            var batch = new List<GenericPowerTelemetry>();
+            foreach(var device in await this.devicesRepository.GetDevicesAsync(cancellationToken))
+            {
+                if (cancellationToken.IsCancellationRequested) break;
+
+                var tokenSecretName = FormatDeviceTokenSecretName(device.SerialNumber);
+                var tokenSecret = await this.keyVaultClient.GetSecretAsync(
+                    this.options.KeyVaultBaseUrl, tokenSecretName, cancellationToken);
+
+                var batterieClient = new SonnenBatterieClient(new TokenCredentials(tokenSecret.Value));
```
Comment (conork20) [inline SmartUtilityAccelerator/SmartUtility/SmartUtility.Core/Sonnen/SonnenBatterieBridge.cs]: Again, batterieClient looks like a typo to me.  Perhaps just use client.

File: SmartUtilityAccelerator/SmartUtility/SmartUtility.Core/Sonnen/SonnenBatterieBridge.cs [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+                var status = await batterieClient.GetStatusAsync(device.SerialNumber, cancellationToken);
+
+                var telemetry = new GenericPowerTelemetry()
+                {
+                    // TODO: extract it from the status.
```
Comment (conork20) [inline SmartUtilityAccelerator/SmartUtility/SmartUtility.Core/Sonnen/SonnenBatterieBridge.cs]: Is this to extract the DateTime from the state received from Sonnen?

File: SmartUtilityAccelerator/SmartUtility/SmartUtility.Core/WattTime/Client/Generated/WattTimeClient.cs [hunk 4]
```diff
@@ -0,0 +1,881 @@
+// <auto-generated>
+// Code generated by Microsoft (R) AutoRest Code Generator.
+// Changes may cause incorrect behavior and will be lost if the code is
+// regenerated.
+// </auto-generated>
```
Comment (conork20) [inline SmartUtilityAccelerator/SmartUtility/SmartUtility.Core/WattTime/Client/Generated/WattTimeClient.cs]: Thanks for this. Could you please add the steps to update this autogenerated code to either (1) the readme of this repo or (2) this OneNote page, which I will add to the readme of this repo? --> https://microsoft.sharepoint.com/teams/Azure_IoT/SolutionsTeam/_layouts/OneNote.aspx?id=%2Fteams%2FAzure_IoT%2FSolutionsTeam%2FSiteAssets%2FSolutionsTeam%20Notebook&wd=target%28Focused%20Solutions%2FSmartUtility.one%7CB588E04C-27AC-4521-BC98-F4262E61E8EE%2FSmartyUtility%20Dev%20Notes%7C7C96F847-0C53-4812-9873-B8EA6205904B%2F%29
onenote:https://microsoft.sharepoint.com/teams/Azure_IoT/SolutionsTeam/SiteAssets/SolutionsTeam%20Notebook/Focused%20Solutions/SmartUtility.one#SmartyUtility%20Dev%20Notes&section-id={B588E04C-27AC-4521-BC98-F4262E61E8EE}&page-id={7C96F847-0C53-4812-9873-B8EA6205904B}&end

That way, if WattTime changes, anyone can update it.

File: SmartUtilityAccelerator/SmartUtility/SmartUtility.Core/Sonnen/Client/Generated/SonnenBatterieClient.cs (hunk 1 above)
Comment (stansw) [reply SmartUtilityAccelerator/SmartUtility/SmartUtility.Core/Sonnen/Client/Generated/SonnenBatterieClient.cs]: I decided to rename it when I realized that the documentation is for "*sonnenBatterie eco 8*". I do want to make it specific as there might be other Sonnen clients we want to include in our solution in the future.

>**1 About this document**
>This document describes the remote control of a sonnenBatterie eco 8

File: SmartUtilityAccelerator/SmartUtility/SmartUtility.Core/WattTime/Client/Generated/WattTimeClient.cs (hunk 4 above)
Comment (stansw) [reply SmartUtilityAccelerator/SmartUtility/SmartUtility.Core/WattTime/Client/Generated/WattTimeClient.cs]: I'll add it to the "Contribute" section

**Pull request merged**